packages/browser/src/index.ts

```diff
@@ -78,6 +83,35 @@ class Browser {
         maskTextClass: maskClass,
       });
     }
+
+    const tracer = opentelemetry.trace.getTracer('@hyperdx/browser');
+
+    if (disableIntercom !== true) {
+      resolveAsyncGlobal('Intercom')
+        .catch()
+        .then(() => {
```

Comment:
nit: Can we just grab Intercom instance from `then` ? I was thinking maybe adding a TS generic type to `resolveAsyncGlobal`

Reply:
That's what I did originally, but I spent a few hours trying to figure out why this code doesn't work (and it's due to trying to use the returned intercom instance).

Afaik Intercom follows a somewhat common pattern of loading a placeholder function under `window.Intercom` while the full script is loading, and then replaces it with the actual function when the intercom script is done loading. This means if you use the intercom reference from the initial page load, you'll be hanging on to a placeholder reference (it'll queue up function calls and flush when the intercom script finally loads), which will no-op one the intercom script is done loading.

Reply:
Ah...replacing the reference. I see. Didn't see the comment. Thanks for the explanation